torch/package/package_importer.py

```diff
@@ -72,13 +73,12 @@ def __init__(
         elif isinstance(file_or_buffer, (Path, str)):
             self.filename = str(file_or_buffer)
             if not os.path.isdir(self.filename):
-                self.zip_reader = torch._C.PyTorchFileReader(self.filename)
+                self.zip_reader = zip_file_reader_type(self.filename)
             else:
-                self.zip_reader = DirectoryReader(self.filename)
+                self.zip_reader = TorchScriptDirectoryReader(self.filename)
         else:
             self.filename = "<binary>"
-            self.zip_reader = torch._C.PyTorchFileReader(file_or_buffer)
-
+            self.zip_reader = TorchScriptPackageZipFileReader(file_or_buffer)
```

Comment:
instead of hardcoding this can we have at the top of the file:

```
HAS_TORCH = False
try:
   import torch
   HAS_TORCH = True
except:
  pass
  
if HAS_TORCH:
  PackageZipFileReader = TorchScriptPackageZipFileReader
else:
  PackageZipFileReader = PythonPackageZipFileReader
```

And then we can just use PackageZipFileReader everywhere and simplify this logic up a lot

Reply:
Instead of this I ended up just doing subclassing for the PackageImporter/Exporter to avoid the fragility of the flags. Is there a reason this approach would be preferable?